src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs

```diff
@@ -46,11 +46,13 @@ public static HandleNullPropagationOption GetDefaultHandleNullPropagationOption(
                 case Linq2SqlQueryProviderNamespace:
                 case ObjectContextQueryProviderNamespaceEF5:
                 case ObjectContextQueryProviderNamespaceEF6:
-                case ObjectContextQueryProviderNamespaceEFCore2:
+             
                     options = HandleNullPropagationOption.False;
                     break;
-
+                
                 case Linq2ObjectsQueryProviderNamespace:
+                // EF Core before 3.0 does a lot of client evaluations and has InMemory support which is the same as Linq2Objects
+                case ObjectContextQueryProviderNamespaceEFCore2: 
```

Comment:
Isn't this a breaking change?

Reply:
Potentially. However, if you try to create expressions like $filter=length(Field) gt 10 when backed by EF InMemory it will throw with version of code that in master now. That's a bug.


When we generate IQueryable  provided by EF  and we have something like 
```C#
x => x.Field.Length > 10
```
 we assumed that EF will pass that operation to the SQL in a form
```SQL
LEN(Field) > 10
```
If Field is null it will; work in SQL, but executed directly C# expression throw exception and we have to convert if to
```C#
x => x.Field == null ? false : x.Field.Length > 10
```
We are doing it for linq to objects, but not EF InMemory

Reply:
@mikepizzo Looks like we don't have tests for $filter and EF Core. Do you want me to:
* Create separate PR to add these tests
* Add these test to this PR
* Create an issue to work on extra EF Core tests after this PR is megred

Reply:
I'm fine with a separate PR for EF Core tests so that we don't hold up this PR any longer -- thanks!